app/github/push_event.rb

```diff
@@ -41,11 +42,16 @@ def deleted?
     event['deleted']
   end
 
-  # The git commit sha to deploy
+  # The git commit sha of this push event.
   def sha
     event['head_commit']['id']
   end
 
+  # Returns true if this push event's git commit sha already has an AutoDeployment.
+  def sha_has_autodeployment?
+    AutoDeployment.exists?(sha: sha)
```

Comment:
So, the uniqueness index is on `(environment_id, sha)`:

https://github.com/remind101/slashdeploy/blob/adea526cbfac35f7b001ae2984bc4f382299800e/db/structure.sql#L485

In other words, it's entirely possible to have multiple auto deployments for the same sha, to different environments (e.g. you could setup CD so that pushing to the `master` branch would deploy to both `prod` and `stage`).

It may be better to solve this by adding a `uniqueness` validation the the AutoDeployment model: http://guides.rubyonrails.org/active_record_validations.html#uniqueness